curvefs/src/client/fuse_s3_client.cpp

```diff
@@ -149,21 +181,31 @@ CURVEFS_ERROR FuseS3Client::FuseOpWrite(fuse_req_t req, fuse_ino_t ino,
         LOG(ERROR) << "UpdateInode fail, ret = " << ret;
         return ret;
     }
+
+    if (fi->flags & O_DIRECT || fi->flags & O_SYNC || fi->flags & O_DSYNC) {
+        // Todo: do some cache flush later
+    }
     return ret;
 }
 
 CURVEFS_ERROR FuseS3Client::FuseOpRead(fuse_req_t req,
-        fuse_ino_t ino, size_t size, off_t off,
-        struct fuse_file_info *fi,
-        char *buffer,
-        size_t *rSize) {
+                fuse_ino_t ino, size_t size, off_t off,
+                struct fuse_file_info *fi,
+                char *buffer, size_t *rSize) {
     Inode inode;
     CURVEFS_ERROR ret = inodeManager_->GetInode(ino, &inode);
     if (ret != CURVEFS_ERROR::OK) {
         LOG(ERROR) << "inodeManager get inode fail, ret = " << ret
                   << ", inodeid = " << ino;
         return ret;
     }
+
+    if (fi->flags & O_DIRECT) {  // check align
+        if (off % DefaultBlksize != 0 || size % DefaultBlksize !=0) {
```

Comment:
you can use this function to check whether a value is aligned to another value which is power of 2

https://github.com/opencurve/curve/blob/a89a10d02d389de1457d7b527701fffb38904802/curvefs/src/space/fast_align.h#L71

Reply:
> you can use this function to check whether a value is aligned to another value which is power of 2
> 
> https://github.com/opencurve/curve/blob/a89a10d02d389de1457d7b527701fffb38904802/curvefs/src/space/fast_align.h#L71

done